sdk/tvOS/OoyalaTVSkinSDK/OOOoyalaTVPlayerViewController.m

```diff
@@ -258,235 +263,253 @@ - (void)hideProgressBar {
+    } else {
+      if (subtitleArray.count == 0) {
+        [subtitleArray addObject:[[OOOoyalaTVPlayerViewController currentLanguageSettings] objectForKey:@"Off"]];
+      }
+      [subtitleArray addObject:language];
     }
-    
-    return [self.tableList copy];
-    
+  }
+
+  // Make the pairs <section name, language/presentation list> to save in the langauge array
+  if (closedCaptionsArray) {
+    Pair *closedCaptionsPair = [Pair alloc];
+    closedCaptionsPair.name = @"Closed Captions";
+    closedCaptionsPair.value = closedCaptionsArray;
+    [self.tableList addObject:closedCaptionsPair];
+  }
+  if (subtitleArray.count > 0) {
+    Pair *subtitlePair = [Pair alloc];
+    subtitlePair.name = [[OOOoyalaTVPlayerViewController currentLanguageSettings] objectForKey:@"Languages"];
+    subtitlePair.value = subtitleArray;
+
+    [self.tableList addObject:subtitlePair];
+  }
+  return [self.tableList copy];
 }
 
 - (void)displayCurrentClosedCaption {
-    if ([self shouldShowClosedCaptions]) {
-        if (self.closedCaptionsView.caption == nil || self.player.playheadTime < self.closedCaptionsView.caption.begin || self.player.playheadTime > self.closedCaptionsView.caption.end) {
-            OOCaption *caption =
-            [self.player.currentItem.closedCaptions captionForLanguage:self.player.closedCaptionsLanguage time:self.player.playheadTime];
-            [self.closedCaptionsView setClosedCaption:caption];
-        }
-    } else {
-        self.closedCaptionsView.caption = nil;
+  if ([self shouldShowClosedCaptions]) {
+    if (!self.closedCaptionsView.caption ||
+        self.player.playheadTime < self.closedCaptionsView.caption.begin ||
+        self.player.playheadTime > self.closedCaptionsView.caption.end) {
+        OOCaption *caption = [self.player.currentItem.closedCaptions captionForLanguage:self.player.closedCaptionsLanguage
+                                                                                   time:self.player.playheadTime];
+        [self.closedCaptionsView setClosedCaption:caption];
     }
+  } else {
+    self.closedCaptionsView.caption = nil;
+  }
 }
 
 - (void)updateClosedCaptionsViewPosition:(CGRect)bottomControlsRect withControlsHide:(BOOL)hidden {
-    CGRect videoRect = [self.player videoRect];
-    if (!hidden) {
-        if (bottomControlsRect.origin.y < videoRect.origin.y + videoRect.size.height) {
-            videoRect.size.height = videoRect.origin.y + videoRect.size.height - bottomControlsRect.size.height;
-        }
+  CGRect videoRect = [self.player videoRect];
+  if (!hidden) {
+    if (bottomControlsRect.origin.y < videoRect.origin.y + videoRect.size.height) {
+      videoRect.size.height = videoRect.origin.y + videoRect.size.height - bottomControlsRect.size.height;
     }
-    [self.closedCaptionsView setFrame:videoRect];
+  }
+  [self.closedCaptionsView setFrame:videoRect];
```

Comment:
I guess it's better to write with dot notation. i.e.
```suggestion
  self.closedCaptionsView.frame = videoRect;
```

Reply:
done